[Review Ready] New station goal: Bluespace mining tap

TDSSS opened the pull request:
**What does this PR do:**
This PR adds a new station goal into rotation, based on an old idea of mine. I've talked to Neca about it and he liked the idea.

**Short Overview:**
So, what is the Bluespace ~~Mining Tap~~ Harvester (renamed)? It is basically a device that reaches into other dimensions thanks to the power of bluespace, mining them for neat stuff. To do so, it requires a LOT of power. Not only that, the more power you supply, the more rewards it generates. This is intended to give engineering a reason to optimize power production, run multiple engines, etc.

**In more details:**
The Power Tap needs to be built on a wire knot, as an apc would not nearly have enough power to keep it supplied. It has 20  mining levels, each mining level requires an order of magnitude more power than the previous one. It is impossible to actually reach the highest level, as far as I can tell.

Depending on your mining level the tap accumulates points, which are spent on generating various benefits. The station goal is to accumulate a reasonably large amount of such points (though you can spend them, they only need to have been earned at some point)

After reworking the reward list, it now provides: Collectable hats, luxury food, as well as various mostly fluffy items with a very small chance for high-tier/rare loot.

At high enough mining levels, there is of course a chance for things to go wrong. What, you thought tapping other dimensions for power and resources was risk-free? The tap has a chance to spawn portals spawning further nasties at a high enough level, so better have sec on stand-by or keep it on a low setting. Emagging it significantly increases the chance of such malfunctions. The portals leave behind bluespace crystals when destroyed, as a small 'reward'.

**The WIP part**
Edit: No longer WIP
- [X] TGUI UI done and prettied up!
- [X]  Sprite got, thanks to Ionward!
- [X] Balancing.  Currently, normal tesla/singulo setup can complete station goal target in ~80 minutes, generate enough points for a cheap item each 15 minutes, and enough for an expensive item in 40 minutes. That's without any real effort with power optimisation.
- [X] Rework the reward list with things heads approve of

**Images of sprite/map changes (IF APPLICABLE):**
Fancy new 3 by 3 sprite, many thanks to Ionward on discord for making it, it's animated too:
![Capture](https://user-images.githubusercontent.com/32099540/63638991-051e3700-c68f-11e9-83ee-da6a4df7d929.PNG)

New and improved TGUI:
Normal operation
The slider has you chose your desired input level, with the bar showing the actual input level it's currently operating at. Buttons should be mostly self-explanatory except for the second one, which decreases desired input level to match actual input level.
![BStap_1](https://user-images.githubusercontent.com/32099540/93764711-e1238100-fc13-11ea-8658-20ecf32e4792.PNG)
The bar turns red if it's unable to reach the desired level due to a lack of power.
![BStap_2](https://user-images.githubusercontent.com/32099540/93764720-e4b70800-fc13-11ea-81d1-ed736edb6ebc.PNG)
If emagged or run with too much power, displays nice scary warnings.
![BStap-3](https://user-images.githubusercontent.com/32099540/93764727-e680cb80-fc13-11ea-9333-0d5496406925.PNG)

Edit: Updated description to account for new reward list.

**Changelog:**
:cl:
add: A new station goal, the bluespace tap. Feed it power for fantastical rewards!
/:cl:

Fox-McCloud commented:
I'm not really keen on something that has the potential to make a an entire job redundant.

Station goals are mostly niche side-projects that provide minor/interesting benefits to the crew that require multiple departments to work together to get up and going--typically in involves cargo, science, and engineering.

The best of these, IMO, is the gene vault--it involves all the prior departments, plus xenobio specifically, botany, and involves most of the crew with running around collecting these samples as well.

This is just an engineering+cargo thing, with the side benefit of "literally just free stuff".

Ideally, whatever is designed should involved multiple departments working together to each fill their own little niche role, and the end result shouldn't invalidate an entire department and should be something that adds a benefit, but isn't such a standout that it changes the round in a dramatic way.

This is also going to likely not play out well when we (inevitably) get techwebs, as one of the anomalies here tied into R&D origin tech.

More station goals are a great idea, but having them be niche, have high crew involvement, and not dramatically alter the round is the best thing to aim for.

TDSSS commented:
It's not supposed to make mining redundant. For one thing it can't produce all minerals, and its intended to be pretty slow in making stuff.

As for involving other departments, sec is involved in fighting off anomalies if the crew is serious about cranking the power level up.

AffectedArc07 commented:
@TDSSS may want to do something like goons PTC like I did a while ago, just do it properly unlike me.
#11050

Necaladun commented:
The actual numbers and what minerals can/can't be brought through are WIP and will need tweaking...it might not even have to be any minerals. Unknown cultural artifact etc is a perfect term - just getting funny hats as a reward for cranking out excess power I think will be popular and fun?

Muhdi345 commented:
Actually making power consumption/generation a factor is great in my book. The station goals are primarily something for Engineering to do if nobody is breaching the station anyway.

I'd love to have some fun with this thing with my SM/TEG setups.

TDSSS commented:
New icons! And yes I know the way I made it a multi-tile machine is hacky but...it's the same as the dna vault

AffectedArc07 commented on code/modules/assembly/signaler.dm:
```diff
@@ -174,3 +174,13 @@
+
+//a random anomaly core, with a random tech level at 7
+/obj/item/assembly/signaler/anomaly/random
+	desc = "A strange anomaly from another universe, who knows what technological discovers it might enable."
```
Technological discoveries maybe?

ktoma36 commented:
I think this is a great idea, it not only gives engineering a reason to build more robust power systems it also supplies them with basic materials for general construction work. One idea I have is to include the ability to pull exotic plants and fauna from different dimensions to supply the chef and botanist with some interesting goodies. Plus it would give an alternative way to supply critters for the DNA vault besides xenobiology spamming gold slime cores.

I could see this leading into some interesting events in the future if it gets merged, I would love to at least give it a try and see it thru til its balanced to tell if the community likes it. This could be something that could be implemented in as a permanent feature, an engineering-science side department (kinda like genetics but without a job specific to it). That's an idea for the future though!

TDSSS commented:
Heads are against it supplying any minerals, so it won't provide engineering with any basic construction material, since then it would be replacing mining. IMO just metal and glass would have been fine but *shrug. I'm working on a big, comprehensive list of things it can give to run past Neca and Kyet right now, feel free to suggest more things here or on discord in coder chat.

TDSSS commented:
I've reworked the rewards list with things heads approved of, now to get the point costs/rewards right. Might create more categories of rewards or add more if you've got ideas.

TDSSS commented:
I've rebalanced point costs. With a no effort tesla or singulo setup, you can get ~600 points per minute. You get the station goal of 50000 points total in ~80 minutes, faster with more power of course. Any further fine tuning will depend on how much power more skilled engineers can squeeze out of the station IMO.

TDSSS commented on code/modules/power/smes.dm:
```diff
@@ -208,8 +208,6 @@
 		disconnect_terminal()
 	return ..()
 
-	return round(5.5*charge/(capacity ? capacity : 5e6))
```
I'm...not sure why/when I did this, but I also took a multi-month break when working on this. I think I just found a useless line and deleted this. Can move this to an unrelated PR if desired.

TDSSS commented on code/modules/station_goals/bluespace_tap.dm:
might be a bit wordy, but I wanted people to know what it is and how it works. Might instead include a 'manual' with the crate? Thoughts?

TDSSS commented on code/modules/station_goals/bluespace_tap.dm:
Not sure this is the right place to do this, but it works out in practise without refactoring station goal completion code.

TDSSS commented on code/modules/station_goals/bluespace_tap.dm:
note to self, remove this lol

TDSSS commented on code/modules/station_goals/bluespace_tap.dm:
I admit I am not sure why loot spawners need a color but copy-pasting for the win.

Couls replied:
if parent has a colour you don't need to set it here, it'll inherit it

TDSSS replied:
Yeah but, why do loot spawners in general have a colour when they're never seen/probably don't have a sprite?

TDSSS commented on code/modules/station_goals/bluespace_tap.dm:
it's harmless, don't worry. Just normal baanas

TDSSS commented on code/modules/station_goals/bluespace_tap.dm:
should probably remove this at some point...

TDSSS commented on code/modules/station_goals/bluespace_tap.dm:
not sure it should actually HAVE an access restriction. Not even the BSA has it, need to remove that comment

TDSSS commented on paradise.dme:
```diff
@@ -224,8 +224,8 @@
 #include "code\controllers\subsystem\nightshift.dm"
 #include "code\controllers\subsystem\npcpool.dm"
 #include "code\controllers\subsystem\overlays.dm"
-#include "code\controllers\subsystem\radio.dm"
```
I have no idea why it did that, I never touched those. Will this be an issue?

Couls replied:
should be fine because its included again one line down

Jovaniph commented:
This new goal would probably make singularity a considerable option (Singularity can put out more power, if see below!).

Atmos Tech can super cool plasma for a shit ton of power in addition, TEG would be more common likely?

TDSSS commented:
I expect multi-engine setups if engineering is serious about getting a 'highscore'

Couls commented on code/modules/station_goals/bluespace_tap.dm:
is this the swarm holopara? seems like this should be a bit rarer

TDSSS replied:
It has a 5 in 168 chance of appearing, so roughly 3%, same for the x-ray injector. With current point balance you can afford to buy this maybe 3 times total if focusing on that and not doing anything super-crazy with power. I think 3 shots at 3% each at a holopara is sufficiently rare. Same for the x-ray.

Couls commented on code/modules/station_goals/bluespace_tap.dm:
xray should also be a bit rarer, if people get their hands on these two it can be a PAIN for security or an antag that has to deal with them

Kyep replied:
Maybe take this out.

Couls commented on code/modules/station_goals/bluespace_tap.dm:
shouldn't this be .tmpl?

TDSSS replied:
I...honestly have no idea why it isn't. Except maybe stupid me read the lowercase L as a capital I and then made a lowercase I or something...

Couls commented on nano/templates/bluespace_tap.tmpi:
```diff
@@ -0,0 +1,60 @@
+<h3>Input Management</h3>
```
should be .tmpl

Drakeven commented:
Perhaps modified to be more of a... Anonymous signature capture, rather than a resource miner would help with not making mining redudant.

TDSSS commented:
I'm not sure what an 'anonymous' signature is, but it was already changed away from giving raw resources, it now has no overlap with mining (except for some things that can also be gotten from abandoned crates)
@Drakeven

TDSSS commented:
Point costs slightly lowered. Since biological artifacts have two chase items (holopara and x-ray), they are now more expensive than cultural items. Might need to add at least 1 very good/desirable cultural item to balance it out a bit.

MrMagolor commented:
>Might need to add at least 1 very good/desirable cultural item to balance it out a bit.

Perhaps alien items of some sort?

Kyep commented:
I like the concept of this (incentive for mining to provide more power).

I very much dislike the implementation, because:
1. What effect will providing stacks of 50 minerals have on mining?
2. What effect will providing luxury food have on the chef?
3. What effect will anomaly cores have on how frequently science makes phazons?

If this was ONLY cosmetic items, that would be fine, cool even.
But when you start including functional items, especially functional items that it is normally up to another job to produce... or which are only rarely available on the station at all... you start running into balance/fun implications of what it does to the rest of the station, and to jobs on the rest of the station.

MrMagolor commented:
>What effect will anomaly cores have on how frequently science makes phazons?

I feel like an anomalous crystal (the colossus drop) would make more sense, while (hopefully) not being as broken.

MrMagolor commented:
Furthermore: having this as a station goal is pretty meh, because you'll only have a 25% chance of rolling it each round. Frankly, I wouldn't mind at all if station shield was removed.

Drakeven commented:
> Furthermore: having this as a station goal is pretty meh, because you'll only have a 25% chance of rolling it each round. Frankly, I wouldn't mind at all if the station shield was removed.

Yes. Please. Station shield is the WORST goal. It needs a re-work... Maybe even just building bigger shields that you actually have to construct or something.

TDSSS commented:
@Kyep 

I'm sorry, it seems you judged this as its initial implementation and not the reworked rewards list. That's partially on me since I didn't fully update the description and just mentioned it in the discussion, but it no longer provides metal or anything like that. Listing it all here would be a very long list but refer to the google doc I sent you a few weeks ago for approval, I marked the things I actually put in with an X.

I've also went ahead and edited the description some more to make it clear now that it doesn't provide raw material anymore. Though it still does provide fancy food, you yourself (and Neca) gave fancy food a yes when I asked, so I'm a bit confused.

It is also my belief that a station goal _should_ provide a tangible benefit. I mean, DNA vault literally can give people x-ray or hulk or something. Meanwhile, station shield does nothing most rounds and crew is understandably very meh on actually building it.

@MrMagolor 
>I feel like an anomalous crystal (the colossus drop) would make more sense, while (hopefully) not being as broken.

>Also, you mention having holopara as a biological thing- if it isn't already, wouldn't scarabs make more sense for that?

It is the scarab thing, I just said holopara because it's the more generally well known term for what it is, IMO. Also, anomaly core has been removed already. What exactly does the anomalous crystal from colossus do?

MrMagolor commented:
I thiiiink you might need to squash commits.

TDSSS commented:
Well, each commit is an actual code change, since it was somewhat of a big project, it's now just all in  arow because I rebased to incorporate changes. But yeah if maints say so I'll squash

MRgaty commented:
But if you can't get any minerals from it, can it be even considered a "mining tap"

TDSSS commented:
It uh...might need to be renamed at some point, yeah. Though it can make alien alloy and brass!

TDSSS commented:
Will resolve conflicts once maints/heads had their discussion, no point in that now when it might accumulate new ones while I wait.

TDSSS commented:
Went ahead and resolved merge conflicts. Also removed the x-ray vision injector, renamed it to Bluespace Harvester and made some minor code changes. Will add more things to replace the x-ray injector later. As for the food I like where it's right now and will keep it like that unless it totally sinks the PR. IMO there already are plenty of ways to get nice food besides the chef.

TDSSS commented:
Added more nice but low-power rewards to compensate for x-ray removal. Namely:
tranquilite and bananium stacks (can't be gotten by mining so no encroaching on another job there)
combat gloves (basically just fancy insulated gloves)
monkey cubes (can be gotten multiple ways already, sometimes chef apparently runs out and needs them for meat)
carpet, both normal and black (for fancy decorating)
extra-special deluxe soap (same soap as captain gets, cleans slightly faster. The envy of drask everywhere)


With that done I think I've addressed every concern raised by Kyet? Any chance this could get a maint review in terms of code or maybe a test merge soon?

taukausanake commented:
> tranquilite and bananium

Hate to break it to you but no one wants the clown running around in a H.O.N.K. mech.

The others I can't comment on

TDSSS commented:
I mean, I could...but they'll be squashed when merged anyway so *shrug

>Hate to break it to you but no one wants the clown running around in a H.O.N.K. mech.

Still need engineering to give them up and then Robotics to make it which needs both RD and Captain approval so...meh

taukausanake commented:
You think Robotics won't make the mech? They'd _love_ to make any mech that isn't a ripley (mining doesn't even want them anymore).
When I say "anyone" I'm really saying "all of the maints and headmins". There's a reason the NO FUN joke is played a bunch. This would be an example. I for one LOVE the idea, I'm just calling them as I see them

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,359 @@
+	lootcount = 1
+
+/obj/effect/spawner/lootdrop/bluespace_tap/hat
+	name = "a head covering"
```
weird wording here. Doesn't flow well. Maybe something like "fashionable headwear"?
or rare headwear etc

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,359 @@
+	var/points = 0	//mining points
+	var/actual_power_usage = 500
+	var/total_points = 0	//total amount of points ever earned, for tracking station goal
+	density = 1
```
TRUE

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,359 @@
+/obj/machinery/power/bluespace_tap/New()
+	..()
+	//more code stolen from dna vault, inculding comment below. Taking bets on that datum being made ever.
+	//TODO: Replace this,bsa and gravgen with some big machinery datum
```
Would be sweet

TDSSS replied:
A bit out of scope though, I think.

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,359 @@
+				input_level = 0	//as hilarious as it would be for the tap to spawn in even more nasties because you can't get to it to turn it off, that might be too much for now. Unless sabotage is involved
+			for(var/i = 1, i <= rand(1, 3), i++)	//freaky shit here, 1-3 freaky portals
+				var/turf/location = locate(x + rand(-5,5), y + rand(-5,5), z)
+				var/obj/structure/spawner/nether/bluespace_tap/portal = new(src)
```
can just new it with location as param instead of src. No need to move it then

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,359 @@
+	var/list/listed_items = list()//a list of lists, each inner list equals a datum
+	for(var/key = 1 to length(product_list))
+		var/datum/data/bluespace_tap_product/A = product_list[key]
+		listed_items.Add(list(list(
```
list of lists of lists? Might have put a list( to much here

TDSSS replied:
It's actually just a list of lists. if you do list.Add and pass it a list, it will add every element of that list to the original list. The documentation says 

> Appends the specified items to the list. If an argument is itself a list, each item in the list will be added. 

So I need to pack it in a second list. You can see the same code in vending.dm where I took it from.

farie82 replied:
Ah just a weird byond thing then. Makes sense...

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,359 @@
+//emaging provides slightly more points but at much greater risk
+/obj/machinery/power/bluespace_tap/emag_act(var/mob/living/user as mob)
+	if(!emagged)
+		emagged = 1
```
TRUE

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,359 @@
+	if(!emagged)
+		emagged = 1
+		if(user)
+			user.visible_message("[user.name] emags the [src.name].","<span class='warning'>You override the safety protocols.</span>")
```
can just use [user] and instead of 'the [src.name]' you can prob use [src]

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,359 @@
+
+
+
+/obj/machinery/power/bluespace_tap/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
```
var implied

TDSSS commented:
Thanks for the others though, will get around to them soon.

TDSSS commented:
Had to rebase to incorporate general codebase changes so it would compile with the new DM version, so had to force push again. But all comments should be incorporated now.

MrMagolor commented:
Once again I would reccomend squashing that long list of commits.

farie82 commented:
> Once again I would reccomend squashing that long list of commits.

Maints can do that upon merge. Authors shouldn't have to worry about it

TDSSS commented:
Line endings should be fixed now.

TDSSS commented:
Let me know when you're debating this and stuff and I'll resolve merge conflicts then.

TDSSS commented:
Note to self: change nanoui if this ever gets looked at again

ZomgPonies commented:
This looks interesting -- Just put in silly non-balance affecting rewards for now just to get it approved for merge, and then it can be balanced down the line with more interesting outputs.

TDSSS commented:
Well, all balance affecting rewards were okayed/pre-approved by the heads at the time, so I would be rather surprised if it fails due to the rewards.

TDSSS commented:
Once #14058 is done, I'll probably copy from there for the TGUI.

TDSSS commented:
TGUI has been significantly improved (see new pictures). With that, I think this is review ready. (obviously it's merge conflicted because of tgui but not worth fixing that atm)

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,380 @@
+	var/list/listed_items = list()//a list of lists, each inner list equals a datum
+	for(var/key = 1 to length(product_list))
+		var/datum/data/bluespace_tap_product/A = product_list[key]
+		listed_items.Add(list(list(	//note that adding a list to a list in BYOND adds the indiviudal list elements
```
You can use this trick:
```
listed_items[++listed_items.len] = your_list
```
This will increase the length of the list by one first and then return the index of the last entry so you can use that

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,380 @@
+
+//produces and vends the product with the desired key
+/obj/machinery/power/bluespace_tap/proc/produce(key)
+	var/datum/data/bluespace_tap_product/A = product_list[key]
```
need to check here if the key falls within the allowed length since it is input from an external source

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,380 @@
+		emagged = TRUE
+		do_sparks(5, FALSE, src)
+		if(user)
+			user.visible_message("[user] emags [src].","<span class='warning'>You override the safety protocols.</span>")
```
Need spans here. Also the text `X emags Y` is a bit short. It can use some more flavor I'd say

TDSSS commented:
Thanks for all those, farie

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -189,42 +188,48 @@
 	new /datum/data/bluespace_tap_product("Unknown Biological Artifact", /obj/effect/spawner/lootdrop/bluespace_tap/organic, 20000)
 	)
 
+	var/input_level = 0	//the level the machine is currently mining at. 0 means off
```
Could you make these valid for the documentation generation tool?
```
/// Info here
var/var_name
```

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,429 @@
+		filler.parent = null
+		qdel(filler)
+	fillers.Cut()
+	. = ..()
```
Just a `return ..()` will do here

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,429 @@
+	var/list/occupied = list()
+	for(var/direct in list(EAST, WEST, SOUTHEAST, SOUTHWEST))
+		occupied += get_step(src, direct)
+	occupied += locate(x+1, y-2, z)
```
spacing here between the operators

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,429 @@
+	if(key <= 0 || key > length(product_list))	//invalid key
+		return
+	var/datum/data/bluespace_tap_product/A = product_list[key]
+	if(!A)
```
Should never happen except if admins (or such) fuck with the internal list

TDSSS replied:
Better safe than sorry, imo.

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,429 @@
+
+/obj/machinery/power/bluespace_tap/tgui_act(action, params)
+	if(..())
+		return 1
```
`TRUE` here if you want to update the screen. Most tgui_acts just return here. They don't return TRUE or any value since if this happens then the UI is not interactive (or not there)

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,429 @@
+
+//UI stuff below
+
+/obj/machinery/power/bluespace_tap/tgui_act(action, params)
```
Looking at other implementations of `tgui_act`. Every implementation returns TRUE if it succeeds. They do this by setting `. = TRUE` just before the action switch.
If the `tgui_act` returns `TRUE` then it means they want to be refreshed. Which I assume you want to happen in the cases below

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,429 @@
+		emagged = TRUE
+		do_sparks(5, FALSE, src)
+		if(user)
+			user.visible_message("<span class='warning'>[user] overrides the safety protocols of [src].</span>","<span class='warning'>You override the safety protocols.</span>")
```
spacing

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,429 @@
+
+//emaging provides slightly more points but at much greater risk
+/obj/machinery/power/bluespace_tap/emag_act(mob/living/user as mob)
+	if(!emagged)
```
Early returns are preferred

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,431 @@
+/obj/machinery/power/bluespace_tap/Destroy()
+	for(var/V in fillers)
+		var/obj/structure/filler/filler = V
+		filler.parent = null
```
Minor thing. Shouldn't this be handled in the Destroy of the filler?

farie82 commented on code/modules/station_goals/dna_vault.dm:
```diff
@@ -136,6 +136,10 @@ GLOBAL_LIST_INIT(non_simple_animals, typecacheof(list(/mob/living/carbon/human/m
 
+/obj/structure/filler/Destroy()
+	parent = null
+	..()
```
Destroy returns a value so this needs to be `return ..()`.
The value returned will determine if the object gets queued for deletion, let alive or hard deleted etc.

farie82 replied:
Still requires the `return ..()`

farie82 reviewed (commented):
Could you also change the implementation in the dna_vault.dm file for the fillers? They currently do the parent removing etc

farie82 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -0,0 +1,429 @@
+		connect_to_network()
+
+/obj/machinery/power/bluespace_tap/Destroy()
+	for(var/V in fillers)
+		qdel(V)
+	fillers.Cut()
```
Can replace this now with `QDEL_LIST(fillers)`

farie82 commented on code/modules/station_goals/dna_vault.dm:
```diff
@@ -188,11 +192,9 @@ GLOBAL_LIST_INIT(non_simple_animals, typecacheof(list(/mob/living/carbon/human/m
 
 /obj/machinery/dna_vault/Destroy()
 	for(var/V in fillers)
```
Same here. `QDEL_LIST`

farie82 reviewed (approved):
LGTM

TDSSS commented:
I've added a paper to the goal crate that contains basic instructions on set-up (especially the part where it needs to be built on a wire knot), since that's something many people might miss.

AffectedArc07 commented on code/modules/station_goals/bluespace_tap.dm:
```diff
@@ -425,3 +425,22 @@
+	name = "paper- 'The Experimental NT Bluespace Harvester - Mining other universes for science and profit!'"
+	info = "<h1>Important Instructions!</h1>Please follow all setup instructions to ensure proper operation. <br>\
+	1. Create a wire knot with ample access to spare power. The device operates independently of APCs. <br>\
+	2. Create a machine frame as normal on the wire knot, taking into account the device's dimensions (3 by 3). <br>\
```
```suggestion
	2. Create a machine frame as normal on the wire knot, taking into account the device's dimensions (3 meters by 3 meters). <br>\
```

If were staying IC with "NT Engineering Standards", give a unit of measure.

variableundefined reviewed (approved):
@TDSSS 

Approved. Please deconflict and prepare for merge. 

My only gripe - along with Affected, is that the sprite is not really fitting in with the rest of the station's style. It looks a bit rounded and the lighting and shading doesn't really match. 

I would love a future sprite that would match our existing machines like the Gravity Generator or BSA better.

variableundefined commented:
@TDSSS Travis failed due to linting issue, try autofixing it.
